omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/FulfillerStatusResource2_2.java

```diff
@@ -0,0 +1,117 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs2_2;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.DateProperty;
+import io.swagger.models.properties.StringProperty;
+import org.openmrs.Order;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.docs.swagger.core.property.EnumProperty;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.annotation.SubResource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link Resource} for Order, supporting standard CRUD operations on fulfiller_comment and fulfiller_status
+ */
+@SubResource(parent = OrderResource2_2.class, path = "fulfillerstatus", supportedClass = Order.class, supportedOpenmrsVersions = {
+		 "2.2.*" })
+
+public class FulfillerStatusResource2_2 extends DelegatingSubResource<FulfillerDetailsRepresentation, Order, OrderResource2_2> {
+	
+	@Override
+	public FulfillerDetailsRepresentation newDelegate() {
+		return new FulfillerDetailsRepresentation();
+	}
+	
+	@Override
+	public FulfillerDetailsRepresentation save(FulfillerDetailsRepresentation delegate) {
+		Order order =  Context.getOrderService().updateOrderFulfillerStatus(delegate.getParent(), delegate.getFulfillerStatus(), delegate.getFulfillerComment());
+		delegate.setParent(order);
+		return  delegate;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription delegatingResourceDescription = new DelegatingResourceDescription();
+		delegatingResourceDescription.addProperty("fulfillerStatus");
+		delegatingResourceDescription.addProperty("fulfillerComment");
+		return delegatingResourceDescription;
+	}
+	
+	@Override
+	public Model getCREATEModel(Representation rep) {
+		return new ModelImpl()
+				.property("fulfillerComment", new StringProperty())
+				.property("fulfillerStatus", new EnumProperty(null));
+	}
+	
+	@Override
+	public Model getUPDATEModel(Representation rep) {
```

Comment:
I think the resource only supports create and not update

Reply:
Are you sure? Because the filler details like comment and status are may updated during the order process. And I am not sure if in this scenario always the Create method would be called by the users of the the REST service. 
Or if maybe at some point some modules would Update these details by submitting it through PUT...?

Reply:
This is a unique resource because the supported class is not persistent so differentiating save vs update doesn't make much sense.